pkg/hooks/manager_test.go

```diff
@@ -129,27 +134,32 @@ var _ = Describe("HooksManager", func() {
 		})
 
 		It("Should find multiple sidecars on different hook points", func() {
-			hookNameMap := map[string]string{
-				"hook1": hooksInfo.OnDefineDomainHookPointName,
-				"hook2": hooksInfo.PreCloudInitIsoHookPointName,
+			hookNameList := []struct {
+				hookName      string
+				hookPointName string
+			}{
+				{"hook1", hooksInfo.OnDefineDomainHookPointName},
+				{"hook2", hooksInfo.PreCloudInitIsoHookPointName},
 			}
-			for hookName, hookPointName := range hookNameMap {
-				socketPath := filepath.Join(socketDir, fmt.Sprintf("%s.sock", hookName))
-				socket, err := hookListenAndServe(socketPath, hookName, hookPointName, 0)
+			for i, hook := range hookNameList {
+				hookPath := filepath.Join(socketDir, "hook-sidecar-"+strconv.Itoa(i))
+				os.MkdirAll(hookPath, os.ModePerm)
+				socketPath := filepath.Join(hookPath, fmt.Sprintf("%s.sock", hook.hookName))
+				socket, err := hookListenAndServe(socketPath, hook.hookName, hook.hookPointName, 0)
 				Expect(err).ToNot(HaveOccurred())
 				defer socket.Close()
 				defer os.Remove(socketPath)
 			}
 
 			manager := newManager(socketDir)
-			err := manager.Collect(uint(len(hookNameMap)), 10*time.Second)
+			err := manager.Collect(uint(len(hookNameList)), 10*time.Second)
 			Expect(err).ToNot(HaveOccurred())
 
 			callbackMaps := manager.CallbacksPerHookPoint
 
-			for _, hookPointName := range hookNameMap {
-				Expect(callbackMaps).Should(HaveKey(hookPointName))
-				Expect(callbackMaps[hookPointName]).Should(HaveLen(1))
+			for _, hook := range hookNameList {
+				Expect(callbackMaps).Should(HaveKey(hook.hookPointName))
+				Expect(callbackMaps[hook.hookPointName]).Should(HaveLen(1))
```

Comment:
Can we compare the actual map to an expected map?
I think it would be easier to follow.

Reply:
lets defer please, i prefer to keep the original subject on the PR, there is no regression, no need cosmetics in this PR